src/installer/tests/HostActivation.Tests/DotnetArgValidation.cs

```diff
@@ -68,6 +68,55 @@ public void MissingArgumentValue_Fails()
                 .And.HaveStdErrContaining($"Failed to parse supported options or their values:");
         }
 
+        [Fact]
+        public void NonManagedFileWithDirectorySeparator_ShowsSpecificError()
+        {
+            // Create a non-.dll/.exe file with directory separator in path
+            string testFile = Path.Combine(sharedTestState.BaseDirectory.Location, "test.txt");
+            File.WriteAllText(testFile, "test content");
+
+            TestContext.BuiltDotNet.Exec(testFile)
+                .CaptureStdOut()
+                .CaptureStdErr()
+                .Execute()
+                .Should().Fail()
+                .And.HaveStdErrContaining($"The application '{testFile}' is not a managed .dll or .exe.");
+        }
```

Comment:
This case can be removed. It is already covered by `FrameworkDependentAppLaunch.Muxer_AssemblyWithDifferentFileExtension_Fails`.